### DISCUSSION TITLE
Welcome to sadl Discussions!

### DISCUSSION THREAD

**Discussion by crapo** [SemanticApplicationDesignLanguage/sadl#586] [General]
@agabaldon , @weisenje , @AbhaMoitra : Discussions are new to GitHub and look like they could be useful so I turned them on.

## 👋 Welcome!
  We’re using Discussions as a place to connect with other members of our community. We hope that you:
  * Ask questions you’re wondering about.
  * Share ideas.
  * Engage with other community members.
  * Welcome others and are open-minded. Remember that this is a community we
  build together 💪.

  For the maintainers, here are some tips 💡 for getting started with Discussions. We'll leave these in Markdown comments for now, but feel free to take out the comments for all maintainers to see.

  📢 **Announce to your community** that Discussions is available! Go ahead and send that tweet, post, or link it from the website to drive traffic here.

  🔗 If you use issue templates, **link any relevant issue templates** such as questions and community conversations to Discussions. Declutter your issues by driving community content to where they belong in Discussions. If you need help, here's a [link to the documentation](https://docs.github.com/en/github/building-a-strong-community/configuring-issue-templates-for-your-repository#configuring-the-template-chooser).

  ➡️ You can **convert issues to discussions** either individually or bulk by labels. Looking at you, issues labeled “question” or “discussion”.

**Comment by pdibona-atl** [comment]
## Translating SADL to OWL without the IDE

My team has been able to use the Eclipse IDE for SADL to generate the .owl files we need for our ontology needs.  However, we need a way to programmatically (from Java or via some REST API) translate  .sadl file(s) into .owl file(s) **without** the IDE.  

At @AbhaMoitra 's recommendation, we looked into the SADL Server, but I do not see a way to do this.  

Would this be possible?  If so, do you have any recommendations for where we could start setting up such a process?

Thanks for you help.

**Comment by crapo** [comment]
To build OWL from SADL files you do not want SadlServer. Rather you want the headless version of the IDE. I have not worked with that but I believe that @tuxji has done so on other projects. It isn't a REST API, but it is a program that you can run standalone to do generation.

**Comment by AbhaMoitra** [comment]
@pdibona-atl : I think there is a bit of confusion. I recommended SadlServer for your internal development. I had said that we do not currently have a way of generating OWL through some command line interface; in the past for ASSERT (which used a SADL Requirements Language built on top of SADL) we did have a command line interface. Hope this clears up the confusion. We do want to have the ability to generate OWL from SADL via a command line interface but that is under consideration at this point.

Thanks @crapo.

**Reply by crapo** [reply]
@tuxji , I thought the headless version provided the capability sought here--to build OWL models from SADL projects programmatically. I thought you had created a headless version for SADL fashioned after the one done for ASSERT(TM). Am I just completely confused?

**Reply by tuxji** [reply]
Yes, SADL has a headless command line application class that had the ability to build OWL models from SADL projects.  That class stopped working last year for some reason, but a project this year will need some way to translate SADL files to OWL files so I'll troubleshoot that class after I help another project prepare for a demo.

**Comment by crapo** [comment]
@tuxji , once the headless application is working I'd like to understand it and add documentation for users desiring to leverage it.

**Comment by pdibona-atl** [comment]
Thank you for the timely and helpful comments.  The headless command-line application sounds like exactly what we need.  As you formally document and roll-out the capability, we would be happy to serve as beta-testers and provide feedback and/or issues we encounter.
